pyproject.toml

```diff
@@ -136,6 +136,7 @@ skip_covered = false
 [tool.pyright]
 reportAny = "none"
 reportExplicitAny = "none"
+reportUnnecessaryCast = "warning"
```

Comment:
Should either also mark it as error in CI or have it an error from the start. 

Letting warnings in will result in polluted basedpyright output, I think we should just treat them as errors eventually.

Reply:
@danielgafni see https://github.com/anam-org/metaxy/pull/181